src/powersoftau/powersoftau.cpp

```diff
@@ -0,0 +1,222 @@
+    po::store(parsed, vm);
+
+    argv0 = argv[0];
+
+    if (vm.count("help")) {
+        help = true;
+        return;
+    }
+
+    if (0 == vm.count("powersoftau_file")) {
+        throw po::error("powersoftau_file not specified");
+    }
+    if (0 == vm.count("degree")) {
+        throw po::error("degree not specified");
+    }
+
+    powersoftau_file = vm["powersoftau_file"].as<std::string>();
+    verbose = vm.count("verbose");
+    verify = vm.count("verify");
+    degree = vm["degree"].as<size_t>();
+    lagrange_degree = vm.count("lagrange-degree")
+                          ? vm["lagrange-degree"].as<size_t>()
+                          : degree;
+    out = vm.count("out") ? vm["out"].as<std::string>()
+                          : "lagrange-" + std::to_string(lagrange_degree);
+    dummy = vm.count("dummy");
+
+    if (dummy && verify) {
+        throw po::error("specify at most one of --dummy and --verify");
+    }
+}
+
+// -----------------------------------------------------------------------------
+// main
+// -----------------------------------------------------------------------------
+
+static int powersoftau_main(const cli_options &options)
+{
+    // Initialize
+    if (options.help) {
+        options.usage();
+        return 0;
+    }
+
+#if 1
+    std::cout << " argv0: " << options.argv0 << std::endl;
+    std::cout << " help: " << std::to_string(options.help) << std::endl;
+    std::cout << " powersoftau_file: " << options.powersoftau_file << std::endl;
+    std::cout << " degree: " << std::to_string(options.degree) << std::endl;
+    std::cout << " verbose: " << std::to_string(options.verbose) << std::endl;
+    std::cout << " verify: " << std::to_string(options.verify) << std::endl;
+    std::cout << " out: " << options.out << std::endl;
+    std::cout << " lagrange_degree: " << std::to_string(options.lagrange_degree)
+              << std::endl;
+#endif
+
+    ppT::init_public_params();
+    if (!options.verbose) {
+        libff::inhibit_profiling_counters = true;
+        libff::inhibit_profiling_info = true;
+    }
+
+    // If --dummy options was given, create a powersoftau struct from
+    // local randomness, write it out and return.
+    if (options.dummy) {
+        const srs_powersoftau<ppT> dummy =
+            dummy_powersoftau<ppT>(options.degree);
+        std::cout << "Writing locally constructed powersoftau to "
+                  << options.powersoftau_file << " ... ";
+        std::ofstream out(
+            options.powersoftau_file,
+            std::ios_base::binary | std::ios_base::out);
+        powersoftau_write(out, dummy);
+
+        std::cout << "DONE" << std::endl;
+        return 0;
+    }
+
+    // Read in powersoftau
+    std::ifstream in(
+        options.powersoftau_file, std::ios_base::binary | std::ios_base::in);
+    const srs_powersoftau<ppT> powersoftau =
+        powersoftau_load(in, options.degree);
+    in.close();
+
+    // If --verify was given, run the verification and stop.
```

Comment:
Just to confirm: this flag checks that the SRS has the right shape, right?
If so, we can keep it that way, but we should remember to rename the function accordingly so that there is no ambiguity with the `verify` algorithm carried out after each contribution (as described in the MPC paper).

Reply:
Renamed this flag to `--check` and renamed the function to `powersoftau_is_well_formed`.